megatron/model/gpt2_model.py

```diff
@@ -271,6 +272,24 @@ def init_specs(self):
                         layer_number=i,
                     )
                 )
+            elif layer_type in ["TE"]:
```

Comment:
needs tested with PP and TP, since we'd be relying on two external codebases (deepspeed for PP, TE for TP) whose topologies probably don't play nicely together.